<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="rviz" default="true"/>

  <!-- Parameters -->
  <arg name="config_file" default="omiya_calibration_room_2023.param.yaml"/>
  <arg name="publish_tfs" default="true"/>
  <arg name="fixed_ground_plane_model" default="false"/>

  <!-- Frame configuration -->
  <arg name="base_frame" default="base_link"/>
  <arg name="main_calibration_sensor_frame" default="pandar_top"/>

  <!-- Lidar configuration -->
  <arg name="calibration_lidar_model" default="pandar64"/>

  <!-- Topic configuration -->
  <arg name="calibration_lidar_topic" default="/sensing/lidar/top/pointcloud_raw"/>

  <arg
    name="auxiliar_tag_ids"
    default="[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49]"
  />
  <arg name="waypoint_tag_ids" default="[0, 1, 2, 3, 4, 5]"/>
  <arg
    name="ground_tag_ids"
    default="[50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99]"
  />
  <arg name="left_wheel_tag_id" default="22"/>
  <arg name="right_wheel_tag_id" default="26"/>

  <!-- tag_based_sfm_base_lidar calibrator -->
  <include file="$(find-pkg-share tag_based_sfm_calibrator)/launch/calibrator.launch.xml">
    <arg name="calibration_service_name" value="calibrate_base_lidar"/>
    <arg name="publish_tfs" value="$(var publish_tfs)"/>
    <arg name="fixed_ground_plane_model" value="$(var fixed_ground_plane_model)"/>

    <arg name="base_frame" value="$(var base_frame)"/>
    <arg name="main_calibration_sensor_frame" value="$(var main_calibration_sensor_frame)"/>
    <arg name="calibration_lidar_0_frame" value="$(var main_calibration_sensor_frame)"/>
    <arg name="calibration_lidar_0_model" value="$(var calibration_lidar_model)"/>
    <arg name="calibration_lidar_0_topic" value="$(var calibration_lidar_topic)"/>

    <arg name="auxiliar_tag_ids" value="$(var auxiliar_tag_ids)"/>
    <arg name="waypoint_tag_ids" value="$(var waypoint_tag_ids)"/>
    <arg name="ground_tag_ids" value="$(var ground_tag_ids)"/>
    <arg name="left_wheel_tag_id" value="$(var left_wheel_tag_id)"/>
    <arg name="right_wheel_tag_id" value="$(var right_wheel_tag_id)"/>
  </include>
</launch>
